expyfun/_input_controllers.py

```diff
@@ -61,19 +65,22 @@ def _clear_keyboard_events(self):
         self.win.dispatch_events()
         self._keyboard_buffer = []
 
-    def _retrieve_keyboard_events(self, live_keys):
+    def _retrieve_keyboard_events(self, live_keys, type='presses'):
         # add escape keys
         if live_keys is not None:
             live_keys = [str(x) for x in live_keys]  # accept ints
             live_keys.extend(self.force_quit_keys)
         self.win.dispatch_events()  # pump events on pyglet windows
         targets = []
+
         for key in self._keyboard_buffer:
-            if live_keys is None or key[0] in live_keys:
-                targets.append(key)
+            if key[2] in self.keyEventTypes[type]:
```

Comment:
I'm worried about the `[2]` index, so just to make sure: `self._keyboard_buffer` will always have 3-element tuples, right?  Because the filtering out of timestamps happens later?  If that's not the case, switching `[2]` to be `[-1]` might make sense.